source/common/config/subscription_factory.h

```diff
@@ -43,6 +44,11 @@ class SubscriptionFactory {
     SubscriptionStats stats = Utility::generateStats(scope);
     switch (config.config_source_specifier_case()) {
     case envoy::api::v2::ConfigSource::kPath:
+      if (!Filesystem::fileExists(config.path())) {
```

Comment:
Technically, in the file case, the file/directory could get deleted between this check and when the watch/load happens, so error handling really should work correctly at that point, or the watch should be setup here. With that said, that is an edge case, and if you want to put in a TODO for that I don't mind. Same below.